tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -0,0 +1,187 @@
+from numbers import Number
+from typing import Tuple, Union
+
+import cv2
+import numpy as np
+import scipy.ndimage as ndi
+import torchvision
+from skimage import exposure, filters, morphology
+from skimage.util import img_as_float
+from torchvision.models._utils import IntermediateLayerGetter
+
+Resolution = Union[Number, Tuple[Number, Number], np.ndarray]
+
+
+def preprocess(fixed_img, moving_img):
+    """This function performs normalization to unify the appearance of fixed
+     and moving images
+
+    Args:
+            fixed_img (:class:`numpy.ndarray`):
+                    A grayscale fixed image
+            moving_img (:class:`numpy.ndarray`):
+                    A grayscale moving image
+
+    Returns:
+            :class:`numpy.ndarray`:
+                    A normalized grayscale fixed image
+            :class:`numpy.ndarray`:
+                    A normalized grayscale moving image
+
+    """
+    if len(fixed_img.shape) != 2 or len(moving_img.shape) != 2:
+        raise ValueError(f'{"The input images should be grayscale images."}')
+
+    moving_entropy, fixed_entropy = filters.rank.entropy(
+        moving_img, morphology.disk(3)
+    ), filters.rank.entropy(fixed_img, morphology.disk(3))
+    if np.mean(fixed_entropy) > np.mean(moving_entropy):
+        moving_img = exposure.match_histograms(moving_img, fixed_img)
+    else:
+        fixed_img = exposure.match_histograms(fixed_img, moving_img)
+
+    return fixed_img, moving_img
+
+
+def dice(mask_1, mask_2):
+    """This function computes dice coefficent between two masks."""
```

Comment:
Missing Args and Returns in the docstring here.

Reply:
Now added. Check here: https://github.com/TissueImageAnalytics/tiatoolbox/blob/dev-dice_metric/tiatoolbox/utils/metrics.py#:~:text=*%20fn)-,def%20dice(gt_mask%2C%20pred_mask)%3A,),-Footer